x-pack/plugins/security_solution/public/common/components/truncatable_text/truncatable_text.tsx

```diff
@@ -25,4 +27,18 @@ export const TruncatableText = styled.span`
     white-space: nowrap;
   }
 `;
-TruncatableText.displayName = 'TruncatableText';
+
+interface Props {
+  showTooltip?: boolean;
+  children: React.ReactNode;
+}
+
+export function TruncatableText({ showTooltip = false, children }: Props) {
+  if (!showTooltip) return <EllipsisText>{children}</EllipsisText>;
```

Comment:
That make sense but this children can be a component with some actions or nested components inside. Maybe allow user to set the tooltip by props instead of use the children could avoid having tooltips with actions or nested components inside.
Something like:
```
interface Props {
  tooltipContent?: React.ReactNode;
  children: React.ReactNode;
}

export function TruncatableText({ tooltipContent, children }: Props) {
  if (!tooltipContent) return <EllipsisText>{children}</EllipsisText>;
  
  return (
    <EuiToolTip display="block" content={tooltipContent}>
      <EllipsisText>{children}</EllipsisText>
    </EuiToolTip>
  );
}
```

What do you think?

Reply:
solid point 👍, I do agree that would make it more flexible. is there a case where children wouldn't just be plain text for this component? it seems like the use case for this component is purely to truncate/add ellipsis to text and passing a component wouldn't work.

Reply:
I just saw [this one](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/formatted_field.tsx#L130) with the text and the tooltip. But as you want, I think it will work in most of the cases. It's just to be more flexible for future approaches 😄

Reply:
somehow missed that one in my search 😅. made the change showTooltip > tooltipContent 😄.